clipping_manager/views.py

```diff
@@ -276,10 +296,17 @@ def get_statistics(self):
         longest_clip = clip_number_of_words[-1] if len(clip_number_of_words) > 0 else ''
         shortest_clip = clip_number_of_words[0] if len(clip_number_of_words) > 0 else ''
 
-        users_with_more_clips = User.objects.exclude(clippings__isnull=True).annotate(models.Count('clippings'))\
-            .filter(clippings__count__gt=clips.count()).count()
+        users_with_more_clips = User.objects.exclude(clippings__isnull=True)\
+                                            .annotate(
+                                                clipings_count=Count(
+                                                    models.Case(models.When(clippings__deleted=False, then=1))
+                                                )
+                                            )\
+                                            .filter(clipings_count__gt=clips.count())\
+                                            .count()            
         clips_rank = users_with_more_clips + 1
 
+        #TODO Filter books with no clippings before comparing book count
```

Comment:
I've tried to figure it out with something like this:
```python
book_count = Subquery(Book.objects.filter(user=OuterRef("pk")).order_by().not_empty()
                            .values("user__pk")
                            .annotate(total=Count("pk"))
                            .values("total"), output_field=IntegerField()
            )
print(User.objects.annotate(count=book_count))
```
but it still doesn't work.

Maybe you have an idea how to get all the books for a specific user, but we only count books that are not empty.

Reply:
Since we're only interested in "Users with more books", we could also just build a query on top of `Book`, filter by not_empty and > the user's books count, and then group by user. and count the result (this should still lead to a `COUNT` in the database I guess, so should be fine).

Reply:
I'm still struggling with this one. I don't quite understand why this query:
```python
books_grouped_by_user = Book.objects.not_empty().values('user_id').annotate(total=Count('user_id')).order_by('total')
```
gives me this result (supposed to be `{'user_id': 2, 'total': 3}, {'user_id': 1, 'total': 83})`:
```
<BookQuerySetManager [{'user_id': 2, 'total': 4}, {'user_id': 1, 'total': 921}]>
```
which means that it sums up the `clippings_count` field for some reason.

Reply:
Strange, your query looks like it's doing what I described above 🤔 Is it the same issue when you use the default manager and queryset instead of `not_empty`?